torch/csrc/Module.cpp

```diff
@@ -2058,6 +2064,51 @@ Call this whenever a new thread is created in order to propagate values from
       },
       "Checks if a tensor's data pointer is COW");
 
+#ifdef USE_CUDA
+#if AT_CUDNN_ENABLED()
+  py_module.def(
+      "_get_cudnn_batch_norm_reserve_space_size",
```

Comment:
Nit - it might be a bit cleaner to add this (private) API unconditionally, but move the macro inside of the imlementation, and in the false case have it raise an error

Reply:
@bdhirsh Reopening this because I could not get the tests to pass without using the macro here. The problem is for non cuda builds we would never load the definition, which lives in `aten/src/ATen/native/cudnn/BatchNorm.cpp`. This led to undefined symbol error for `_get_cudnn_batch_norm_reserve_space_size` when `import * from torch._C`. It may be because that file is guarded by the `AT_CUDNN_ENABLED()` macro, which I believe is only available on cuda builds. I'm adding back the macros here for now but please let me know if you can think of a better way to get rid of them.

Reply:
Another option is to guard on the macro directly in the lambda you put below:

```
      [](const at::Tensor& input) {
#if AT_CUDNN_ENABLED()
        return at::native::_get_cudnn_batch_norm_reserve_space_size(input);
#else
        TORCH_CHECK(false, "PyTorch was not built with cudnn");
      },
```

Reply:
Ok, let me give that a try